tests/unit/data_formats/arrow/test_arrow_format.py

```diff
@@ -0,0 +1,331 @@
+# Copyright (C) 2023 Intel Corporation
+#
+# SPDX-License-Identifier: MIT
+
+
+import os
+import platform
+from functools import partial
+
+import numpy as np
+import pytest
+
+from datumaro.components.dataset_base import DatasetItem
+from datumaro.components.environment import Environment
+from datumaro.components.importer import DatasetImportError
+from datumaro.components.media import Image
+from datumaro.components.project import Dataset
+from datumaro.plugins.data_formats.arrow import ArrowExporter, ArrowImporter
+from datumaro.plugins.transforms import Sort
+
+from ....requirements import Requirements, mark_requirement
+
+from tests.utils.test_utils import check_save_and_load, compare_datasets, compare_datasets_strict
+
+
+class ArrowFormatTest:
+    exporter = ArrowExporter
+    importer = ArrowImporter
+    format = ArrowImporter.NAME
+
+    def _test_save_and_load(
+        self,
+        helper_tc,
+        source_dataset,
+        converter,
+        test_dir,
+        target_dataset=None,
+        importer_args=None,
+        compare=compare_datasets_strict,
+        **kwargs,
+    ):
+        def post_hook(save_dir, *args, **kwargs):
+            # TODO: This is a hacky solution. Need to take a look into it.
+            # python on windows in github action can not remove arrow files
+            # with an access denied error but files can be removable from CMD.
+            # It might be related to not closed arrow writer object
+            # though some files are removable.
+            if platform.system() != "Windows":
+                return
+            for file in os.listdir(save_dir):
+                if not file.endswith(".arrow"):
+                    continue
+                os.system(f"rm -rf {os.path.join(save_dir, file)}")
```

Comment:
I think you can resolve this in more `pytest`-onic way by overriding `test_dir` fixture.

Reply:
Thanks for the suggestion. Fixed in https://github.com/openvinotoolkit/datumaro/pull/931/commits/b3f8104d1a52d77df9e9836139aa91488bb8df2f